fuel-vm/src/gas.rs

```diff
@@ -124,3 +136,361 @@ impl GasUnit {
         Self::Accumulated(self.cost() + other.cost())
     }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+/// Gas costings for every op.
+/// The inner values are wrapped in an [`Arc`]
+/// so this is cheap to clone.
+pub struct GasCosts(Arc<GasCostsValues>);
+
+impl GasCosts {
+    /// Create new cost values wrapped in an [`Arc`].
+    pub fn new(costs: GasCostsValues) -> Self {
+        Self(Arc::new(costs))
+    }
+}
+
+impl Default for GasCosts {
+    fn default() -> Self {
+        Self(Arc::new(GasCostsValues::default()))
+    }
+}
+
+impl Default for GasCostsValues {
+    fn default() -> Self {
+        // The default values for gas costs
+        // are generated from fuel-core-benches.
+        default_gas_costs::default_gas_costs()
+    }
+}
+
+#[allow(missing_docs)]
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(default = "GasCostsValues::unit"))]
+/// Gas costs for every op.
+pub struct GasCostsValues {
+    pub add: Word,
+    pub addi: Word,
+    pub aloc: Word,
+    pub and: Word,
+    pub andi: Word,
+    pub bal: Word,
+    pub bhei: Word,
+    pub bhsh: Word,
+    pub burn: Word,
+    pub cb: Word,
+    pub cfei: Word,
+    pub cfsi: Word,
+    pub croo: Word,
+    pub div: Word,
+    pub divi: Word,
+    pub ecr: Word,
+    pub eq: Word,
+    pub exp: Word,
+    pub expi: Word,
+    pub flag: Word,
+    pub gm: Word,
+    pub gt: Word,
+    pub gtf: Word,
+    pub ji: Word,
+    pub jmp: Word,
+    pub jne: Word,
+    pub jnei: Word,
+    pub jnzi: Word,
+    pub k256: Word,
+    pub lb: Word,
+    pub log: Word,
+    pub lt: Word,
+    pub lw: Word,
+    pub mcpi: Word,
+    pub mint: Word,
+    pub mlog: Word,
+    #[cfg_attr(feature = "serde", serde(rename = "mod"))]
+    pub mod_op: Word,
+    pub modi: Word,
+    #[cfg_attr(feature = "serde", serde(rename = "move"))]
+    pub move_op: Word,
+    pub movi: Word,
+    pub mroo: Word,
+    pub mul: Word,
+    pub muli: Word,
+    pub noop: Word,
+    pub not: Word,
+    pub or: Word,
+    pub ori: Word,
+    #[cfg_attr(feature = "serde", serde(rename = "ret_contract"))]
+    pub ret: Word,
+    #[cfg_attr(feature = "serde", serde(rename = "rvrt_contract"))]
+    pub rvrt: Word,
+    pub s256: Word,
+    pub sb: Word,
+    pub scwq: Word,
+    pub sll: Word,
+    pub slli: Word,
+    pub srl: Word,
+    pub srli: Word,
+    pub srw: Word,
+    pub sub: Word,
+    pub subi: Word,
+    pub sw: Word,
+    pub sww: Word,
+    pub swwq: Word,
+    pub time: Word,
+    pub tr: Word,
+    pub tro: Word,
+    pub xor: Word,
+    pub xori: Word,
+
+    // Dependant
```

Comment:
I know changing this would require a lot of changes, so just a comment, but we may want to consider the conventions of American vs Australian vs British English for comments and naming, would be good to add to the rfcs

Reply:
It seems like @adlerjohn has preferred using American English at least in the specs.

Reply:
I can just do a grep and changes this. I didn't realize it was spelt differently in the US.